tests/helpers/util.py

```diff
@@ -84,13 +84,13 @@ def run_agent_with_fake_backend(config_text, fake_services):
         output = io.BytesIO()
 
         def pull_output():
-            if sys.platform == 'win32':
-                b, _ = proc.communicate()
-                output.write(b)
-                return
             while True:
                 # If any output is waiting, grab it.
-                ready, _, _ = select.select([proc.stdout], [], [], 0)
+                if sys.platform != 'win32':
```

Comment:
Since this is run in a separate thread I think it can actually just block on the read below and doesn't need any of the select/ready logic.

Reply:
Agreed.